lib/SILOptimizer/Transforms/CopyForwarding.cpp

```diff
@@ -352,7 +355,9 @@ class AnalyzeForwardUse
     return true;
   }
   bool visitStoreInst(StoreInst *Store) {
-    llvm_unreachable("illegal reinitialization or store of an address");
+    // Trivial values may be stored prior to the next deinit. A store is an
+    // implicit "deinit" with no operand to replace.
```

Comment:
Useful to assert if the type  `isTrivial` before returning `true` ?

Reply:
Yes, that assert would actually be fine. I added it but need to wait for the SCK testing. The real problem diagnosing some of these asserts is that they're implemented as "unreachable". We should not use "unreachable" to check for unexpected SIL patterns.